app/UuidModel.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace App;
+
+use Illuminate\Database\Eloquent\Model;
+
+use Ramsey\Uuid\Uuid;
+
+class UuidModel extends Model
+{
+    public $incrementing = false;
+
+    protected $keyType = 'string';
+
+    public function __construct(array $attributes = [])
+    {
+        parent::__construct($attributes);
+    }
+
+    public static function boot()
+    {
+        parent::boot();
+
+        self::creating(function ($model) {
+            $model->{$model->getKeyName()} = Uuid::uuid4()->toString();
```

Comment:
In contrast to other UUID versions, UUID4 is designed specifically to be random, which is why it's used here. The intention herein is for this value to be very difficult to guess, so as to prevent malicious user-agents from hijacking other users' orders.

Reply:
You are definitely more qualified to handle anything related to UUID as I'm not familiar with it, so LGTM :laughing: